### ISSUE TITLE
Google Home Developer Console 

### ISSUE THREAD

**Issue by y8s** [mikejac/node-red-contrib-google-smarthome#597]
I got this email. Will the changes affect how this service works?

 

> Hi usernameredacted@gmail.com,
> 
> ‌
> 
> Thanks for building smart home Actions with Google on the Actions console. We’re excited to announce that the new [Google Home Developer Console](https://console.home.google.com/projects) will soon replace the Actions console as the destination to develop and launch your projects.
> 
> ‌
> 
> Your Actions projects will migrate to the Developer Console in stages, and will not impact your users or your product experience. It’s important to make sure your Actions and the Simulator are set up correctly before the migration. Beyond that, no action is required. Please see the [migration FAQ](https://developers.home.google.com/cloud-to-cloud/project/migration#do_i_need_to_modify_my_smart_home_projects_or_actions_to_prepare_for_the_migration) for details.
> 
> 
> Key dates
> 
> Starting in early November
> You can no longer create new Actions on the Action console – all new projects will be built on the Developer Console. You can edit existing projects until the migration starts.
> Certification approval and launch requests from the Actions console may be rejected due to project migration. Please refrain from submitting these requests unless it’s critical.
> Around mid November
> All Cloud and Local Home SDK projects will start migrating to the Developer Console in batches. You’ll receive an email when your projects have been migrated.
> ‌
> 
> To access migrated projects on the Developer Console, you’ll need to agree to the Terms of Service for Google Home Developers.
> 
> ‌
> 
> We’ll send a reminder email about two weeks before the migration date.
> 
> ‌
> 
> Best,
> 
> The Google Home Developer team

**Comment by Caprico85** [comment]
Hello,

I got the same email. So far I don't know any more details. I can only hope that projects will be migrated automatically and that this "will not impact your users or your product experience" as Google promises.

But to be honest, I'm not sure what to make of it. I migrated one of my projects and created a new one. None of them have the necessary settings (such as the fulfilment URL) to make a smarthome service work. There also seems to be no documentation available. I'm curious to see if they really will have the new Dev Console ready by mid-November when they migrate all the projects. It seems a bit rushed. I expect the transition to be at least "a bit" rough.

Anyway, I'll keep you updated if I get any news.

**Comment by FireWizard52** [comment]
Hi all,

Today I received the following email from Google.

I have no idea why Google thinks, that this should be confidential

> **CONFIDENTIAL - Do not forward**
> 
> Hi  <email address deleted>
> 
> ‌We'd like to remind you that your existing Actions on Google smart home  
> projects will start migrating to the Google Home Developer Console in  
> batches starting around early December.
> 
> This will not impact your users or your product experience. It's important  
> to make sure your Actions and the Simulator are set up correctly before the  
> migration. Please see the Actions migration FAQ for proper setup details.  
> Beyond that, no action is required.
> 
> Key dates
> 
> Starting in mid November
> 
> You can no longer create new Actions on the Action console – all new  
> projects will be built on the Developer Console. You can edit existing  
> projects until the migration starts.
> Certification approval and launch requests from the Actions console may be  
> rejected due to project migration. Please refrain from submitting these  
> requests unless it's critical.
> 
> Around early December
> 
> All Cloud and Local Home SDK projects will start migrating to the Developer  
> Console in batches. You'll receive an email when your projects have been  
> migrated.
> 
> To access your migrated projects on the Developer Console, you'll need to  
> agree to the Terms of Service for Google Home Developers.
> 
> Best,
> 
> The Google Home Developer team
> Google LLC 1600 Amphitheatre Parkway, Mountain View, CA 94043.
> 
> This email was sent to <email deleted> because you showed interest in  
> developing Actions on Google.

**Comment by y8s** [comment]
I tried to create an action today and you can't. the smart home option in google actions is grayed out and unselectable. it only provides a link to the developer console.

**Comment by y8s** [comment]
I haven't tested for this node but I did get the Google assistant integration for home assistant working via the new Google home developer console using a cloud to cloud integration.

Most of the same stuff is there but all on one page. you just stop after saving and skip the test portion.

**Comment by WouterGritter** [comment]
I also can't create a Smart Home project. The "Start building" button is greyed out :(

**Comment by Caprico85** [comment]
For me, it doesn't work either.

I can migrate my existing project to the Developer Console. But then I get the message that it has to be managed on the old Actions Console.

![image](https://github.com/user-attachments/assets/bb90a0a2-a038-444d-9f1b-8112ce38bcc3)

If I create a new project, I can click the button "Add integration" on the Develop page. But it only throws me back to the Resources page. On the Resources page, there are only some tutorials. There's nothing where I can set up my smart home project.

![image](https://github.com/user-attachments/assets/1dca558e-3c51-4684-9923-d5f52352d1ba)

**Comment by y8s** [comment]
OK try this walkthrough:
# Create Project in Google Home Developer Console
1. Go to the [Google Home Developer Console](https://console.home.google.com/projects)
2. Click on *Create a Project*
![Screenshot 2024-11-22 141432](https://github.com/user-attachments/assets/0a6f9501-4447-4bf6-bc47-55f669949767)
4. On the Get Started page, click *Create a Project*
![Screenshot 2024-11-22 141824](https://github.com/user-attachments/assets/f3a5374b-2174-4d53-962d-4fc0b265dbc4)
5. Enter a project name and click *Create Project* 
![9cf67b392989a51075faa1b52f312cf8](https://github.com/user-attachments/assets/cba89ca6-f56c-4554-886b-a38dd34a589b)
6. On the *Project Home* page, add a Cloud-to-Cloud integration.
![Screenshot 2024-11-22 142124](https://github.com/user-attachments/assets/b7d4541a-6fc8-4156-b753-9d0b6baea0cb)
**(This step seems to be available for me immediately after creating a project but not ALWAYS.)**

7. Click the *Next: Develop* button at the bottom.
8. Click the *Next: Setup* button at the bottom.
9. Enter a name for the integration.
10. Select all applicable device types (you can select them all).
11. Create an app icon 144 x 144 pixels and upload it by selecting the *Upload image* button. 
12. under *Account Linking*, in the *OAuth Client ID* field, add your project ID. Save this for use later.
13. Add a 32 character string from a password generator (without special characters) to the *Client Secret* field. Save this for use later.
15. Enter `https://yourdomain:port/oauth` in the *Authorization URL* field.
16. Enter `https://yourdomain:port/token` in the *Token URL* field.
17. Enter `https://yourdomain:port/smarthome` in the *Cloud fulfillment URL* field.
18. Skip *Local fulfillment*, *App Flip*, and *Scope(s)*  for now and leave the "Have Google transmit Client ID and secret via HTTP basic auth header" **UN**checked.
19. Click *Save*
20. You will see your project saved with the status "Draft".
21. You do not need to test.

Continue from existing instructions at [Enable HomeGraph API](https://github.com/mikejac/node-red-contrib-google-smarthome/blob/master//docs/setup_instructions.md#enable-homegraph-api)

**Comment by Caprico85** [comment]
@y8s, a very big thank you!!! On the Dev Console, I have always used the links in the left sidebar menu. But the Setup section is missing from this menu. It's only accessible via the "Next: Setup" button, which I never clicked. This was exactly what I was looking for.

This finally gives me the chance to update the setup instructions for the new Dev console. Can I use your walkthrough as a starting point?

**Comment by y8s** [comment]
>  Can I use your walkthrough as a starting point?

Absolutely! I based it on the existing [setup instructions](https://github.com/mikejac/node-red-contrib-google-smarthome/blob/master/docs/setup_instructions.md) when making my screenshots. But I got lazy and didn't make them for the rest of the process.

**Comment by FireWizard52** [comment]
Hi,

It is probably a good idea to update the Setup instructions. 
But I think it is better to wait until Google has finished the migration process and that you have received the email confirmation.

I'm not sure, but if you create a new project now, you might end up with a duplicate project.
To keep the smarthome working, I don't think, we have to do anything now, as far I have read.

**Comment by y8s** [comment]
> I'm not sure, but if you create a new project now, you might end up with a duplicate project. To keep the smarthome working, I don't think, we have to do anything now, as far I have read.

Possibly but you could just create a new one with a different name and ID and ignore the old one when it migrates.

**Comment by FireWizard52** [comment]
Maybe, but there is no reason to create a new similar project, unless you want to create a Setup instruction for the Developer console with corresponding screenshots.

I expect to get a correct Setup after the migration by Google. In that case also local fulfillment will work 
We will see.

**Comment by FireWizard52** [comment]
Hi all,

Yesterday evening, I received the following email from Google:
(Again confidential)

> **CONFIDENTIAL - Do not forward**
> 
> Hi email address
> ‌
> Your existing Actions on Google smart home projects have been successfully  
> migrated to the Developer Console! There was no impact on these Actions or  
> the user experience. You can now modify these Actions and create new  
> integrations in this new console.
> 
> If you have questions about the migration or the console, feel free to  
> contact your account manager or provide feedback on this form.
> 
> Best,
> 
> The Google Home Developer team

Today I had the opportunity to explore the "Developer  Console".
Although things are new, I was immediately directed to my project and was able to find the settings.
All devices are working correct and also "local fulfillment" is working as expected.

So it seems the migrations does not cause any issues for the users.

Regards

**Comment by y8s** [comment]
got the same email. the only real issue is you have to have an app icon or you can't save/make changes.

**Comment by Caprico85** [comment]
I got this mail too. Everything still works. I didn't expect the migration to run this well.

Now I only need to finish my updated setup guide.

**Comment by Caprico85** [comment]
The setup instructions were updated to the Dev Console now.
Instructions for local fulfillment still need to be done.

**Comment by Caprico85** [comment]
All instructions are updated now.

**Comment by y8s** [comment]
Thanks @Caprico85 !
